exo-setup/src/app-setup.spec.ls

```diff
@@ -0,0 +1,102 @@
+require! {
+  '../../exosphere-shared': {run-process}
+  'chai': {expect}
+  'fs-extra' : fs
+  'js-yaml' : yaml
+  'mkdirp'
+  'path'
+  'prelude-ls': {each}
+  'rimraf'
+}
+
+
+describe 'AppSetup', ->
+
+  describe 'set up a complex app' (...) ->
+    @timeout 600000
+
+    before (done) ~>
+      @app-name = 'complex-setup-app'
+      @app-dir = path.join process.cwd!, 'tmp', @app-name
+      @docker-compose-location = path.join @app-dir, 'tmp', 'docker-compose.yml'
+      @exocom-name = 'exocom0.22.1'
+      @internal-services = ['html-server', 'todo-service']
+      @external-services = ['external-service']
+      @external-dependencies = ['mongo3.4.0']
+      @process = checkout-setup-app @app-name, @app-dir
+        ..on 'ended' done
+
+    specify 'should create docker-compose.yml at the expected location', (done) ~>
+      fs.stat @docker-compose-location, (err, stat) ~>
+        expect(err).to.be.null
+        @docker-compose = yaml.safe-load fs.read-file-sync(@docker-compose-location, 'utf8')
+        done!
+
+    specify 'should list all services and dependencies under \'services\'', (done) ~>
```

Comment:
Even though the `before` method is asynchronous? This should be the expected behavior: https://stackoverflow.com/questions/24723374/async-function-in-mocha-before-is-alway-finished-before-it-spec